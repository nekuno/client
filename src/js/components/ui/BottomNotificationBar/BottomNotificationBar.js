import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './BottomNotificationBar.scss';
import ProgressBar from "../ProgressBar/ProgressBar";
import connectToStores from "../../../utils/connectToStores";
import WorkersStore from "../../../stores/WorkersStore";
import translate from "../../../i18n/Translate";

function getState() {
    const linksPercentage = WorkersStore.getLinksPercentage();

    return {
        linksPercentage
    };
}

@translate('BottomNotificationBar')
@connectToStores([WorkersStore], getState)
export default class BottomNotificationBar extends Component {

    static propTypes = {
        // Injected by @translate:
        strings: PropTypes.object,
        //Injected by @connectToStores
        linksPercentage      : PropTypes.number,
    };

    render() {
        const {children, strings, linksPercentage} = this.props;
        let displayStyle = null;
        switch (linksPercentage) {
            case null:
                displayStyle = styles.hidden;
                break;
            case 100:
                displayStyle = styles.fadeOut;
                break;
            default:
                ((displayStyle !== styles.fadeIn) ? displayStyle = styles.fadeIn : '');
                break;
        }
        return (
            <div className={displayStyle}>
                <div className={styles.bottomNotificationBar}>
                    <div className={styles.bottomNotificationBarText}>{strings.analyzingData}</div>
                    <div className={styles.bottomNotificationBarProgressbar}>
                        <ProgressBar percentage={linksPercentage} textColor={'#F0F1FA'} size={'small'} background={'#756EE5'} strokeColor={'#FFF'}/>
                    </div>
                </div>
            </div>
        );
    }
}

BottomNotificationBar.defaultProps = {
    strings: {
        analyzingData   : 'Analyzing data...',
    }
};