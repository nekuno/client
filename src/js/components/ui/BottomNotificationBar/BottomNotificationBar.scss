@import '../../../../scss/_partials/colors';

.bottom-notification-bar {
  background: $purple-main;
  bottom: 0;
  color: $gray-border;
  font-size: 14px;
  position: sticky;
  width: 90%;
  margin: 0px auto 10px auto;
  padding: 0 10px;



  border-radius: 5px 5px 5px 5px;
  -moz-border-radius: 5px 5px 5px 5px;
  -webkit-border-radius: 5px 5px 5px 5px;
  border: 0px solid #000000;

  .bottom-notification-bar-text {
    display: inline-block;
    width: 50%;
    text-align: left;
  }
  .bottom-notification-bar-progressbar {
    display: inline-block;
    width: 50%;
    color: $gray-border;
  }

}