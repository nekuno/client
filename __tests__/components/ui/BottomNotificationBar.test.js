import React from 'react';
import BottomNotificationBar from '../../../src/js/components/ui/BottomNotificationBar/BottomNotificationBar.js';
import styles from '../../../src/js/components/ui/BottomNotificationBar/BottomNotificationBar.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test BottomNotificationBar component', () => {
    const bottomNotificationBar = shallow(<BottomNotificationBar>Lorem ipsum</BottomNotificationBar>);

    it('should be defined', () => {
        expect(bottomNotificationBar).toBeDefined();
    });
    it('should render correctly', () => {
        expect(bottomNotificationBar).toMatchSnapshot();
    });
});
