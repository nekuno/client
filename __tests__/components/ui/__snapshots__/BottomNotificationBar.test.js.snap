// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BottomNotificationBar component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <connectToStores(BottomNotificationBar)
    strings={
      Object {
        "analyzingData": "Analyzing data...",
      }
    }
  >
    Lorem ipsum
  </connectToStores(BottomNotificationBar)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Lorem ipsum",
      "linksPercentage": null,
      "strings": Object {
        "analyzingData": "Analyzing data...",
      },
    },
    "ref": null,
    "rendered": "Lorem ipsum",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Lorem ipsum",
        "linksPercentage": null,
        "strings": Object {
          "analyzingData": "Analyzing data...",
        },
      },
      "ref": null,
      "rendered": "Lorem ipsum",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
